pkg/utils/security/validation.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2023 The Fluid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package security
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const safePathFormat string = "[\\.\\-_\\/A-Za-z0-9]+"
+const safePathErrMsg string = "path: %s doesn't pass the validation, a legal path can only contains alphanumeric characters or characters: '-', '_', '/', '.'"
+
+var safePathRegexp = regexp.MustCompile("^" + safePathFormat + "$")
```

Comment:
Can you reuse  util function `validation.IsDNS1035Label` ?

Reply:
Maybe it cannot be simply reused. In the implementation of `validation.IsDNS1035Label`, it also uses regexp to validate input string, but the regexp is too strict (e.g. `_`, `A-Z` is not allowd, can only start with `a-z`...). We can reuse `validation.IsDNS1035Label` by spliting the path and validate each directory name (still need to handle `_`, `A-Z` and so on, e.g. turn the directory name to lower case), but I think it is unnecessary.

Reply:
1. How about  separating the path into its individual components using the directory separator. This approach effectively mitigates potential issues arising from scenarios like ///.

```golang
func IsValidMountRootPath(path string) bool {
    // Normalize the path and split it into components
    parts := strings.Split(filepath.Clean(path), string(filepath.Separator))

    for _, part := range parts {
        // Convert characters to lowercase and replace underscores with hyphens
        part = strings.ToLower(part)
        part = strings.Replace(part, "_", "-", -1)

        // If the component fails the DNS 1123 conformity test, the function returns false
        if len(IsDNS1123Label(part)) > 0 {
            return false
        }
    }

    // If all components pass the DNS 1123 check, the function returns true
    return true
}
```

2. Relocate the function into  validation package. Put this function in the security package may create a false impression about its actual use.

Reply:
I agree with the point 2. For the point 1, the only concern is the DNS 1035 validation is too strict for users (e.g. if some user's mount root is /opt/20-runtime-mnt, it will not pass the validation) and after users update Fluid, the application previously running fine crushes now.